### PR TITLE
Configurable openshift component logging

### PR DESCRIPTION
**Pull request by jim-minter**
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/specs/fakerp/loglevel.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	var (
+		azurecli *azure.Client
+		cli      *standard.SanityChecker
+		ctx      = context.Background()
+	)
+
+	BeforeEach(func() {
+		var err error
+		azurecli, err = azure.NewClientFromEnvironment(true)
+		Expect(err).NotTo(HaveOccurred())
```
Comment (charlesakalugwu) [inline test/e2e/specs/fakerp/loglevel.go]: add `Expect(azurecli).NotTo(BeNil())`

**Comment by charlesakalugwu** [comment]
@jim-minter had a look at this and it looks generally good to me. I am not sure how the hack in `pkg/config/generate.go` is an improvement but i'm curious on how you propose we move forward.

File: test/e2e/specs/fakerp/loglevel.go (hunk 1 above)
Comment (jim-minter) [reply test/e2e/specs/fakerp/loglevel.go]: Our tests grow a log when we add lines like these - I am happy to, and would rather, code on the assumption that if no error is returned, the return value is non-nil.

**Pull request merged**